### PR TITLE
convert to custom vars from where

### PR DESCRIPTION
**Pull request by shawn-hurley**

### CODE REVIEW

File: pkg/conversion/convert.go [hunk 1]
```diff
@@ -123,7 +125,7 @@ func convertWindupWhenToAnalyzer(windupWhen windup.When, where map[string]string
 	if windupWhen.And != nil {
 		whens := []map[string]interface{}{}
 		for _, condition := range windupWhen.And {
-			converted := convertWindupWhenToAnalyzer(condition, where)
+			converted, _ := convertWindupWhenToAnalyzer(condition, where)
```
Comment (fabianvf) [inline pkg/conversion/convert.go:128]: Should we drop it here or does it just not matter?

File: pkg/conversion/convert.go [hunk 2]
```diff
@@ -339,7 +342,27 @@ func convertWindupWhenToAnalyzer(windupWhen windup.When, where map[string]string
...
-	return conditions
+	return conditions, customVars
+}
+
+func convertWhereToCustomVars(whereMap map[string]string, fullPattern string) []map[string]interface{} {
+	l := []map[string]interface{}{}
+	newString := fullPattern
+	for k, v := range whereMap {
+		newString = strings.ReplaceAll(newString, "{"+k+"}.", fmt.Sprintf("(?P<%s>%s.)?", k, v))
+		newString = strings.ReplaceAll(newString, "{"+k+"}", fmt.Sprintf("(?P<%s>%s)?", k, v))
```
Comment (fabianvf) [inline pkg/conversion/convert.go:353]: This won't ever match that pattern though right, since we already replaced it in the line above?

Comment (shawn-hurley) [reply pkg/conversion/convert.go:353]: no, because the `.` at the end there does some extra magic :)

File: pkg/conversion/convert.go (hunk 1 above)
Comment (shawn-hurley) [reply pkg/conversion/convert.go:128]: It does not matter right now, I don't see a place where we have to deal with it 

we would have to handle the cases where there are collisions, and I don't see a need right at this moment as I don't think they exist. 

Thoughts?

**Pull request merged**